### PR TITLE
Plat 606 superset jsreport refactor

### PR DESCRIPTION
**Pull request by nour-borgi**
In this PR - addressed Jsreport and Superset Packages - : 

- exit if the deploy fails

+ I added a healthcheck inside docker-compose instead of await-helper to do checks automatically for both cases (dev mount true or not)

Before it was showing a success message even though the jsreport was failing:
image

now with the healthcheck:
image

### CODE REVIEW

File: dashboard-visualiser-jsreport/docker-compose.yml [hunk 1]
```diff
@@ -3,6 +3,12 @@ version: '3.9'
 services:
   dashboard-visualiser-jsreport:
     image: jsreport/jsreport:3.2.0
+    healthcheck:
+      test: wget --no-verbose --tries=1 --spider -q http://localhost:5488/ || exit 1
```
Comment (marrouchi) [inline dashboard-visualiser-jsreport/docker-compose.yml:7]: Isn't there a healthceck endpoint for jsreport ?

File: dashboard-visualiser-jsreport/swarm.sh [hunk 2]
```diff
@@ -1,71 +1,117 @@
...
+  readonly ACTION
+  readonly MODE
+  readonly COMPOSE_FILE_PATH
+  readonly ROOT_PATH
+  readonly service_name
+}
+
+# shellcheck disable=SC1091
+function import_sources() {
+  source "${ROOT_PATH}/utils/docker-utils.sh"
```
Comment (marrouchi) [inline dashboard-visualiser-jsreport/swarm.sh:31]: Let's declare a "UTILS_PATH" var

File: utils/docker-utils.sh [hunk 3]
```diff
@@ -181,12 +182,24 @@ docker::deploy_service() {
         docker_compose_dev="-c ${DOCKER_COMPOSE_PATH}/$DOCKER_COMPOSE_DEV_FILE"
     fi
 
+    if [[ -n "${DOCKER_COMPOSE_DEV_MOUNT}" ]]; then
+        docker_compose_dev="$docker_compose_dev -c ${DOCKER_COMPOSE_PATH}/$DOCKER_COMPOSE_DEV_MOUNT"
+    fi
```
Comment (marrouchi) [inline utils/docker-utils.sh:187]: Can we add a comment here ?

**Review by marrouchi — CHANGES_REQUESTED** [review]

File: utils/docker-utils.sh (hunk 3 above)
Comment (nour-borgi) [reply utils/docker-utils.sh]: Yes sure, done in [d6a6ac1](https://github.com/jembi/platform/pull/191/commits/d6a6ac163363b2bd066cf25e6f00ee0034836cce)

File: dashboard-visualiser-jsreport/docker-compose.yml (hunk 1 above)
Comment (nour-borgi) [reply dashboard-visualiser-jsreport/docker-compose.yml]: Yep you're right there is an endpoint, updated it in [533796a](https://github.com/jembi/platform/pull/191/commits/533796ad0cc26c446131335675ffb2a1fbeab4a3)

File: dashboard-visualiser-superset/swarm.sh [hunk 4]
```diff
@@ -1,67 +1,91 @@
...
-. "${ROOT_PATH}/utils/docker-utils.sh"
-. "${ROOT_PATH}/utils/log.sh"
+function initialize_package() {
+  local superset_dev_compose_param=""
 
-main() {
   if [[ "${MODE}" == "dev" ]]; then
     log info "Running Dashboard Visualiser Superset package in DEV mode"
-    superset_dev_compose_param="-c ${COMPOSE_FILE_PATH}/docker-compose.dev.yml"
+    superset_dev_compose_param="docker-compose.dev.yml"
```
[the remark this answers is not in the document]
Comment (nour-borgi) [reply dashboard-visualiser-superset/swarm.sh]: cool done in [fad605b](https://github.com/jembi/platform/pull/191/commits/fad605bda9f8793cfe2803c151bd3e1577a6399f)

File: dashboard-visualiser-jsreport/swarm.sh [hunk 5]
```diff
@@ -1,71 +1,117 @@
...
   fi
+}
+
+function initialize_package() {
+  local js_report_dev_compose_param=""
+  local js_report_dev_mount_compose_param=""
+
+  if [[ "${MODE}" == "dev" ]]; then
+    log info "Running Dashboard Visualiser Jsreport package in DEV mode"
+    js_report_dev_compose_param="docker-compose.dev.yml"
```
[the remark this answers is not in the document]
Comment (nour-borgi) [reply dashboard-visualiser-jsreport/swarm.sh]: cool done in [fad605b](https://github.com/jembi/platform/pull/191/commits/fad605bda9f8793cfe2803c151bd3e1577a6399f)

File: dashboard-visualiser-jsreport/swarm.sh (hunk 2 above)
Comment (nour-borgi) [reply dashboard-visualiser-jsreport/swarm.sh]: cool done in [fad605b](https://github.com/jembi/platform/pull/191/commits/fad605bda9f8793cfe2803c151bd3e1577a6399f)

File: dashboard-visualiser-jsreport/swarm.sh [hunk 6]
```diff
@@ -1,71 +1,117 @@
...
+  readonly service_name
+}
+
+# shellcheck disable=SC1091
+function import_sources() {
+  source "${UTILS_PATH}/docker-utils.sh"
+  source "${UTILS_PATH}/log.sh"
+}
 
 unbound_ES_HOSTS_check() {
```
Comment (marrouchi) [inline dashboard-visualiser-jsreport/swarm.sh:35]:
```suggestion
check_es_hosts_env_var() {
```

File: dashboard-visualiser-superset/swarm.sh [hunk 7]
```diff
@@ -1,67 +1,91 @@
...
   fi
 
-  if [[ "${ACTION}" == "init" ]] || [[ "${ACTION}" == "up" ]]; then
-    config::set_config_digests "$COMPOSE_FILE_PATH"/docker-compose.yml
-    try "docker stack deploy -c ${COMPOSE_FILE_PATH}/docker-compose.yml $superset_dev_compose_param instant" "Failed to deploy Dashboard Visualiser Superset"
+  (
+    docker::deploy_service "${COMPOSE_FILE_PATH}" "docker-compose.yml" "$superset_dev_compose_filename"
+    docker::deploy_sanity "${service_name}"
+  ) || {
+    log error "Failed to deploy Dashboard Visualiser Superset package"
```
Comment (marrouchi) [inline dashboard-visualiser-superset/swarm.sh:50]:
```suggestion
    log error "Failed to deploy Dashboard Visualizer Superset package"
```

File: dashboard-visualiser-superset/swarm.sh [hunk 8]
```diff
@@ -1,67 +1,91 @@
...
-    config::await_service_removed instant_superset-config-importer
+main() {
+  init_vars "$@"
+  import_sources
 
-    log info "Removing stale configs..."
-    config::remove_stale_service_configs "$COMPOSE_FILE_PATH"/importer/docker-compose.config.yml "superset"
-    config::remove_stale_service_configs "$COMPOSE_FILE_PATH"/docker-compose.yml "superset"
+  if [[ "${ACTION}" == "init" ]] || [[ "${ACTION}" == "up" ]]; then
+    log info "Running Dashboard Visualiser Superset package in ${NODE_MODE} node mode"
```
Comment (marrouchi) [inline dashboard-visualiser-superset/swarm.sh:78]:
```suggestion
    log info "Running Dashboard Visualizer Superset package in ${NODE_MODE} node mode"
```

File: dashboard-visualiser-superset/swarm.sh [hunk 9]
```diff
@@ -1,67 +1,91 @@
...
 
-    docker::deploy_sanity dashboard-visualiser-superset
+    initialize_package
   elif [[ "${ACTION}" == "down" ]]; then
-    try "docker service scale instant_dashboard-visualiser-superset=0" "Failed to scale down dashboard-visualiser-superset"
-  elif [[ "${ACTION}" == "destroy" ]]; then
-    docker::service_destroy dashboard-visualiser-superset
-    docker::service_destroy superset-config-importer
+    log info "Scaling down Dashboard Visualiser Superset"
```
Comment (marrouchi) [inline dashboard-visualiser-superset/swarm.sh:82]:
```suggestion
    log info "Scaling down Dashboard Visualizer Superset"
```

File: dashboard-visualiser-superset/swarm.sh [hunk 10]
```diff
@@ -1,67 +1,91 @@
...
 
-    # Removing Superset volumes
-    docker::try_remove_volume superset
-    docker::try_remove_volume superset-frontend
-    docker::try_remove_volume superset_home
+    scale_services_down
+  elif [[ "${ACTION}" == "destroy" ]]; then
+    log info "Destroying Dashboard Visualiser Superset"
```
Comment (marrouchi) [inline dashboard-visualiser-superset/swarm.sh:86]:
```suggestion
    log info "Destroying Dashboard Visualizer Superset"
```

**Review by marrouchi — CHANGES_REQUESTED** [review]

File: utils/docker-utils.sh [hunk 11]
```diff
@@ -230,16 +231,22 @@ docker::deploy_service() {
...
+    # Adding Dev compose file to the params
     if [[ -n "${DOCKER_COMPOSE_DEV_FILE}" ]]; then
-        docker_compose_dev="-c ${DOCKER_COMPOSE_PATH}/$DOCKER_COMPOSE_DEV_FILE"
+        docker_compose_param="-c ${DOCKER_COMPOSE_PATH}/$DOCKER_COMPOSE_DEV_FILE"
+    fi
+
+    # Adding Dev mount compose file to the params
+    if [[ -n "${DOCKER_COMPOSE_DEV_MOUNT}" ]]; then
+        docker_compose_param="$docker_compose_param -c ${DOCKER_COMPOSE_PATH}/$DOCKER_COMPOSE_DEV_MOUNT"
     fi
```
Comment (marrouchi) [inline utils/docker-utils.sh:242]: Let's do if else, and invert the conditions, wdyt ?

File: dashboard-visualiser-jsreport/swarm.sh [hunk 12]
```diff
@@ -1,71 +1,117 @@
...
-ROOT_PATH="${COMPOSE_FILE_PATH}/.."
-. "${ROOT_PATH}/utils/config-utils.sh"
-. "${ROOT_PATH}/utils/docker-utils.sh"
-. "${ROOT_PATH}/utils/log.sh"
+  COMPOSE_FILE_PATH=$(
+    cd "$(dirname "${BASH_SOURCE[0]}")" || exit
+    pwd -P
+  )
+
+  UTILS_PATH="${COMPOSE_FILE_PATH}/../utils"
```
Comment (marrouchi) [inline dashboard-visualiser-jsreport/swarm.sh:18]: Since this code is repeating in my files, wdyt about refactoring ?

**Review by marrouchi — CHANGES_REQUESTED** [review]

File: dashboard-visualiser-jsreport/swarm.sh (hunk 12 above)
Comment (marrouchi) [reply dashboard-visualiser-jsreport/swarm.sh:18]: Or should it be done in a seperate PR ?

File: utils/docker-utils.sh (hunk 11 above)
Comment (nour-borgi) [reply utils/docker-utils.sh:242]: they can both exist, we can have DEV and DEV_MOUNT compose files

File: dashboard-visualiser-jsreport/swarm.sh (hunk 12 above)
Comment (nour-borgi) [reply dashboard-visualiser-jsreport/swarm.sh]: sure I will write this down with the other notes and made it in another PR if it is possible in Bash

File: dashboard-visualiser-jsreport/swarm.sh (hunk 6 above)
Comment (nour-borgi) [reply dashboard-visualiser-jsreport/swarm.sh]: done in [40901f1](https://github.com/jembi/platform/pull/191/commits/40901f1adeeed343fe505f55b697103a097d4d76)

**Review by marrouchi — APPROVED** [review]

**Pull request merged**